utils/layer.py

```diff
@@ -53,32 +53,46 @@ def add_attributes(layer, assets):
     layer.updateFields()
 
 
-def add_features(layer, assets, geometry_type):
-    """
-    Function to add features to the layer.
-
-    Arguments:
-        qgis_layer layer: This is a layer in QGIS.
-        list assets: This is a list of assets (GeoJSONs).
-    """
-    # Create the features
+def add_features(layer, list_of_assets):
+    """Function to add features to the layer."""
+    # Create the feature(s)
     layer.startEditing()
     features = []
-    for asset in assets:
+    for asset in list_of_assets:
         feature = QgsFeature(layer.pendingFields())
         geometry = asset.pop("geometry")
-        if geometry_type is "Point":
+        if str(geometry["type"]) == "Point":
```

Comment:
I think it's better to refactor this geometry part a bit. For example, you can put this part in a function that returns a QgsGeometry, and then you can set the geometry. It is generally a good idea to separate logic.

 ```python
for asset in list_of_assets:
    ...
    qgs_geometry = geometry_from_dict(geometry)
    feature.setGeometry(qgs_geometry)
    ...
```